### ISSUE TITLE
vega system not working 

### ISSUE THREAD

**Issue by mwcraig** [sncosmo/sncosmo#205]
Like in #174, the vega system seems to be unavailable. Using the code from that example,

```python
import sncosmo

model = sncosmo.Model(source='salt2')
print model.bandmag('desi','vega',0.)
```

generates a timeout error. Subsequent attempts generate a FITS error (empty or corrupt FITS file) instead.

The traceback below is from the initial timeout error:

```
Downloading http://sncosmo.github.io/data/bandpasses/des/des_i.dat
|========================================================================| 4.1k/4.1k (100.00%)         0s
---------------------------------------------------------------------------
URLError                                  Traceback (most recent call last)
<ipython-input-3-8d6cbb91b748> in <module>()
----> 1 print model.bandmag('desi','vega',0.)

/Users/mcraig/anaconda/lib/python2.7/site-packages/sncosmo/models.pyc in bandmag(self, band, magsys, time)
   1431             The return value is an `~numpy.ndarray` if any are interable.
   1432         """
-> 1433         return _bandmag(self, band, magsys, time)
   1434
   1435     def color(self, band1, band2, magsys, time):

/Users/mcraig/anaconda/lib/python2.7/site-packages/sncosmo/models.pyc in _bandmag(model, band, magsys, time_or_phase)
    183     result = np.empty(bandflux.shape, dtype=np.float)
    184     for i, (b, ms, f) in enumerate(zip(band, magsys, bandflux)):
--> 185         ms = get_magsystem(ms)
    186         zpf = ms.zpbandflux(b)
    187         result[i] = -2.5 * np.log10(f / zpf)

/Users/mcraig/anaconda/lib/python2.7/site-packages/sncosmo/magsystems.pyc in get_magsystem(name)
     23     if isinstance(name, MagSystem):
     24         return name
---> 25     return _MAGSYSTEMS.retrieve(name)
     26
     27

/Users/mcraig/anaconda/lib/python2.7/site-packages/sncosmo/_registry.pyc in retrieve(self, name, version)
    160             func, args, meta = self._loaders[key]
    161             if version is None:
--> 162                 self._instances[key] = func(*args, name=name)
    163             else:
    164                 self._instances[key] = func(*args, name=name, version=version)

/Users/mcraig/anaconda/lib/python2.7/site-packages/sncosmo/builtins.pyc in load_spectral_magsys_fits(relpath, name)
    708
    709 def load_spectral_magsys_fits(relpath, name=None):
--> 710     abspath = DATADIR.abspath(relpath)
    711     hdulist = fits.open(abspath)
    712     dispersion = hdulist[1].data['WAVELENGTH']

/Users/mcraig/anaconda/lib/python2.7/site-packages/sncosmo/utils.pyc in abspath(self, relpath, isdir)
    472             else:
    473                 url = self._get_url(relpath)
--> 474                 download_file(url, abspath)
    475
    476         return abspath

/Users/mcraig/anaconda/lib/python2.7/site-packages/sncosmo/utils.pyc in download_file(remote_url, local_name)
    320         try:
    321             with open(local_name, 'wb') as target:
--> 322                 _download_file(remote_url, target)
    323         except HTTPError:
    324             # in case of error downloading, remove opened file.

/Users/mcraig/anaconda/lib/python2.7/site-packages/sncosmo/utils.pyc in _download_file(remote_url, target)
    267             e.reason.strerror = msg
    268             e.reason.args = (e.reason.errno, msg)
--> 269         raise e
    270
    271     # This isn't supposed to happen, but occasionally a socket.timeout gets

URLError: <urlopen error ftp error: timed out>
```

ping @janeglanzer who discovered the bug...

**Comment by kbarbary** [comment]
Seems OK now. Maybe the STScI FTP site was just down for a bit?
```
In [1]: import sncosmo

In [2]: ms = sncosmo.get_magsystem("vega")
Downloading ftp://ftp.stsci.edu/cdbs/calspec/alpha_lyr_stis_007.fits
|=========================================| 241k/241k (100.00%)         0s
```
The download code is supposed to add the requested URL to the error message so the user can more easily check if the server is just temporarily down, but I guess the conditions [here](https://github.com/sncosmo/sncosmo/blob/de4e165a04d8363bd12822f9042b669222a72582/sncosmo/utils.py#L262-L263) were not satisfied in this case of an FTP address.

**Comment by mwcraig** [comment]
I tracked down the issue. Here the download takes ~7 seconds, and the timeout, read from the astropy config file, is 3 seconds by default. Not sure there is much you can do about that, though an easy way to change the timeout would be helpful.

A related, but somewhat separate, issue is that you should probably delete the attempted download if the request times out. Right now an empty file is created, so subsequent attempts fail because there is an empty file in the local cache.

**Comment by kbarbary** [comment]
Thanks for tracking this down Matt. I pushed some fixes in #206.
